src/main.rs

```diff
@@ -147,13 +180,134 @@ impl App {
     }
 
     fn save_sample(data: &[i16], path: &Path, name: &str, sample_type: &str) -> Result<()> {
-        let output = normalize_path(path, name)?;
+        let output = normalize_path(path, name, "wav")?;
         write_sample_to_file(data, &output)?;
         let space = if sample_type.is_empty() { "" } else { " " };
         println!("Wrote {sample_type}{space}sample to {output:?}");
 
         Ok(())
     }
+
+    fn get_sample_memory_backup(&mut self) -> Result<BackupData> {
+        let volca = self.volca()?;
+
+        let mut backup = BackupData::new();
+
+        for header in volca
+            .iter_sample_headers()
+            .filter(|res| res.as_ref().map_or(true, |header| !header.is_empty()))
+        {
+            let header = header?;
+            backup.sample_slots[header.sample_no as usize] = Some(header.name);
+        }
+
+        Ok(backup)
+    }
+
+    fn download_backup_data(&mut self, output: PathBuf) -> Result<()> {
+        let backup = self.get_sample_memory_backup()?;
+        Self::save_backup_data(backup, output)
+    }
+
+    fn save_backup_data(backup: BackupData, output: PathBuf) -> Result<()> {
+        let f = fs::OpenOptions::new()
+            .write(true)
+            .create(true)
+            .open(&output)?;
+        serde_yaml::to_writer(f, &backup)?;
+
+        Ok(())
+    }
+
+    fn load_backup_data(input: &PathBuf) -> Result<BackupData> {
+        // check extension to enforce yaml format
+        let ext = match input.extension() {
+            Some(ffi_str) => ffi_str.to_str().unwrap_or(""),
+            None => "",
+        };
+
+        if ext != "yaml" {
```

Comment:
I believe `yml` is also a valid extension. I think you can remove this check and let serde do its job.